diff -ur mars_orig/emutli1.c mars_new/emutli1.c
--- mars_orig/emutli1.c	1998-05-07 05:32:52.000000000 -0400
+++ mars_new/emutli1.c	2021-01-03 00:17:17.870362263 -0500
@@ -33,7 +33,8 @@
 #include <sys/time.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/config.h>
+//#include <linux/config.h>
+#include <linux/ipx.h>
 #if 0
 # include <linux/sockios.h>
 #endif
@@ -340,7 +341,7 @@
     errorp(10, "Problem", "probably kernel-IPX is not setup correctly");
     exit(1);
   } else {
-    ipx_config_data cfgdata;
+    struct ipx_config_data cfgdata;
     struct sockaddr_ipx ipxs;
     ioctl(sock, SIOCIPXCFGDATA, &cfgdata);
     org_auto_interfaces =
diff -ur mars_orig/makefile.unx mars_new/makefile.unx
--- mars_orig/makefile.unx	2003-06-14 11:08:40.000000000 -0400
+++ mars_new/makefile.unx	2021-01-03 01:16:16.405418594 -0500
@@ -66,7 +66,7 @@
 $(DESTMAKEFILE): $(VPATH)/makefile.unx $(VPATH)/config.h
 	rm -f $@
 	cp $(VPATH)/makefile.unx makefile.c
-	$(CPP) $(HOSTCFLAGS) -I$(VPATH) makefile.c > $@
+	$(CPP) $(HOSTCFLAGS) -I$(VPATH) makefile.c | sed 's/^    /\t/g' | sed 's/^ /\t/g' > $@
 	rm -f makefile.c
 #endif
 
@@ -125,9 +125,7 @@
 
 OBJ1= $(EMUTLIOBJ) net1$(O) tools$(O)
 OBJ2= $(OBJ1) $(EMUTLIOBJ1) $(NWROUTE_O)
-OBJ3= $(OBJ1) connect$(O) namspace$(O) nwvolume$(O) nwfile$(O) unxfile$(O) \
-nwqconn$(O) nameos2$(O) nwfname$(O) nwshare$(O) extpipe$(O) \
-nwattrib$(O) trustee$(O)
+OBJ3= $(OBJ1) connect$(O) namspace$(O) nwvolume$(O) nwfile$(O) unxfile$(O) nwqconn$(O) nameos2$(O) nwfname$(O) nwshare$(O) extpipe$(O) nwattrib$(O) trustee$(O)
 OBJ4= $(OBJ1)
 OBJ5= $(OBJ1)
 OBJ6= $(OBJ1) nwdbm$(O) nwcrypt$(O) unxlog$(O) sema$(O) nwqueue$(O) unxfile$(O)
@@ -135,17 +133,9 @@
 OBJ8= $(OBJ6)
 OBJ9= tools$(O) nwfname$(O) unxfile$(O) nwvolume$(O) nwattrib$(O) trustee$(O)
 
-OBJS= $(EMUTLIOBJ) net1$(O) tools$(O) \
-	  $(EMUTLIOBJ1) $(NWROUTE_O) \
-	  connect$(O) namspace$(O) nwvolume$(O) nwfile$(O) unxfile$(O)\
-	  nwqconn$(O) nameos2$(O) nwfname$(O) nwshare$(O) extpipe$(O) \
-	  nwattrib$(O) trustee$(O) \
-	  nwdbm$(O) nwcrypt$(O) unxlog$(O) sema$(O) nwqueue$(O)  \
-	  $(PROG2)$(O) $(PROG3)$(O) $(PROG4)$(O) $(PROG5)$(O) $(PROG6)$(O) \
-	  $(PROG7)$(O) $(PROG8)$(O) $(PROG9)$(O)
+OBJS= $(EMUTLIOBJ) net1$(O) tools$(O) $(EMUTLIOBJ1) $(NWROUTE_O) connect$(O) namspace$(O) nwvolume$(O) nwfile$(O) unxfile$(O) nwqconn$(O) nameos2$(O) nwfname$(O) nwshare$(O) extpipe$(O) nwattrib$(O) trustee$(O) nwdbm$(O) nwcrypt$(O) unxlog$(O) sema$(O) nwqueue$(O) $(PROG2)$(O) $(PROG3)$(O) $(PROG4)$(O) $(PROG5)$(O) $(PROG6)$(O) $(PROG7)$(O) $(PROG8)$(O) $(PROG9)$(O)
 
-HOBJ3= $(PROG3)$(O) connect$(O) namspace$(O) nwvolume$(O) nwfile$(O) \
-	   unxfile$(O)
+HOBJ3= $(PROG3)$(O) connect$(O) namspace$(O) nwvolume$(O) nwfile$(O) unxfile$(O)
 
 HOBJ6= $(PROG6)$(O) sema$(O)
 
@@ -158,7 +148,7 @@
 	   $(CC) -o $(VPATH)/$(PROG2) $(PROG2)$(O) $(OBJ2) $(CRYPTLIB) $(NSLLIB)
 
 #if 0
-#$(CC) -o $(VPATH)/$(PROG2) $(PROG2)$(O) $(OBJ2) $(NDBMLIB) $(CRYPTLIB) $(NSLLIB)
+#$(CC) -o $(VPATH)/$(PROG2) $(PROG2)$(O) $(OBJ2) -lgdbm -lcrypt $(CRYPTLIB) $(NSLLIB)
 #endif
 
 $(PROG3):  $(PROG3)$(O) $(OBJ3)
@@ -171,13 +161,13 @@
 	   $(CC) -o $(VPATH)/$(PROG5) $(PROG5)$(O) $(OBJ5) $(CRYPTLIB) $(NSLLIB)
 
 $(PROG6):  $(PROG6)$(O) $(OBJ6)
-	   $(CC) -o $(VPATH)/$(PROG6) $(PROG6)$(O) $(OBJ6) $(NDBMLIB) $(CRYPTLIB) $(NSLLIB)
+	   $(CC) -o $(VPATH)/$(PROG6) $(PROG6)$(O) $(OBJ6) -lgdbm -lcrypt $(CRYPTLIB) $(NSLLIB)
 
 $(PROG7):  $(PROG7)$(O) $(OBJ7)
 	   $(CC) -o $(VPATH)/$(PROG7) $(PROG7)$(O) $(OBJ7) $(CRYPTLIB) $(NSLLIB)
 
 $(PROG8):  $(PROG8)$(O) $(OBJ8)
-	   $(CC) -o $(VPATH)/$(PROG8) $(PROG8)$(O) $(OBJ8) $(NDBMLIB) $(CRYPTLIB) $(NSLLIB)
+	   $(CC) -o $(VPATH)/$(PROG8) $(PROG8)$(O) $(OBJ8) -lgdbm -lcrypt $(CRYPTLIB) $(NSLLIB)
 
 $(PROG9):  $(PROG9)$(O) $(OBJ9)
 	   $(CC) -o $(VPATH)/$(PROG9) $(PROG9)$(O) $(OBJ9) $(CRYPTLIB) $(NSLLIB)
@@ -189,8 +179,7 @@
 tools$(O): $(DESTMAKEFILE)
 
 $(C)$(O):
-	$(CC) -c $(CFLAGS) $(HOSTCFLAGS)\
- -D_VERS_H_=$(V_H) -D_VERS_L_=$(V_L) -D_VERS_P_=$(P_L) $<
+	$(CC) -c $(CFLAGS) $(HOSTCFLAGS) -DUSE_GDBM -D_VERS_H_=$(V_H) -D_VERS_L_=$(V_L) -D_VERS_P_=$(P_L) $<
 
 n_all:   $(PROGS)
 	 @echo "don't forget to do a 'make install' as root !" >> $(VPATH)/.mk.notes
@@ -205,47 +194,9 @@
 n_install:
 	cd $(VPATH)  && $(INSTALL) -d $(M_PATHNAME_PROGS) && cd $(OBJDIR)
 	cd $(VPATH)  && $(INSTALL) $(INSTALLPROGS) $(M_PATHNAME_PROGS) && cd $(OBJDIR)
-	@-if [ -r /sbin/nwserv ] && [ -r /usr/sbin/nwserv ] ; then \
-		  echo "remove old version in /sbin ?" ; \
-		  (cd /sbin && rm -i nwserv nwbind ncpserv nwconn nwclient nwrouted) ; \
-		fi ;
-	@-if [ -r /etc/nwobj.pag ] ; then \
-		  ( \
-	  if [ ! -r $(M_PATHNAME_BINDERY) ] ; then \
-		   if $(VPATH)/tools.sh yesno "move bindery files from /etc to $(M_PATHNAME_BINDERY) ?" ; then \
-			 mkdir -p $(M_PATHNAME_BINDERY); \
-			 (cd /etc && mv \
-			 nwiobj.pag nwiobj.dir nwobj.pag nwobj.dir \
-			 nwprop.pag nwprop.dir nwval.pag nwval.dir \
-			 $(M_PATHNAME_BINDERY) ) ; \
-		   fi; \
-		  fi; \
-		  ) \
-		fi ;
-	@cd $(VPATH) && (if [ -r $(M_FILENAME_NW_INI) ] ; then \
-echo ""; \
-echo "********************************************************"; \
-echo ""; \
-echo "$(M_FILENAME_NW_INI) exist."; \
-echo "to overwrite it with nw.ini, make install_ini";\
-echo ""; \
-echo "********************************************************"; \
-echo ""; \
-else \
-$(INSTALL) -m 664 nw.ini $(M_FILENAME_NW_INI); \
-echo ""; \
-echo "********************************************************"; \
-echo ""; \
-echo "$(M_FILENAME_NW_INI) created from nw.ini"; \
-echo ""; \
-echo "********************************************************"; \
-echo ""; \
-fi; \
-echo "If you have problems, please read 'doc/BUGS' before"; \
-echo "sending mail to mailinglist or me."; \
-echo ""; \
-echo "********************************************************"; \
-cd $(OBJDIR) )
+	@-if [ -r /sbin/nwserv ] && [ -r /usr/sbin/nwserv ] ; then echo "remove old version in /sbin ?" ; (cd /sbin && rm -i nwserv nwbind ncpserv nwconn nwclient) ; fi;
+	@-if [ -r /etc/nwobj.pag ] ; then (if [ ! -r $(M_PATHNAME_BINDERY) ] ; then if $(VPATH)/tools.sh yesno "move bindery files from /etc to $(M_PATHNAME_BINDERY) ?" ; then mkdir -p $(M_PATHNAME_BINDERY); (cd /etc && mv nwiobj.pag nwiobj.dir nwobj.pag nwobj.dir nwprop.pag nwprop.dir nwval.pag nwval.dir $(M_PATHNAME_BINDERY) ) ; fi; fi; ) fi;
+	@cd $(VPATH) && (if [ -r $(M_FILENAME_NW_INI) ] ; then echo ""; echo "********************************************************"; echo ""; echo "$(M_FILENAME_NW_INI) exist."; echo "to overwrite it with nw.ini, make install_ini"; echo ""; echo "********************************************************"; echo ""; else $(INSTALL) -m 664 nw.ini $(M_FILENAME_NW_INI); echo ""; echo "********************************************************"; echo ""; echo "$(M_FILENAME_NW_INI) created from nw.ini"; echo ""; echo "********************************************************"; echo ""; fi; echo "If you have problems, please read 'doc/BUGS' before"; echo "sending mail to mailinglist or me."; echo ""; echo "********************************************************"; cd $(OBJDIR))
 
 n_reboot: n_install
 	  -(nwserv -k ; nwserv)&
@@ -254,44 +205,20 @@
 	cd $(VPATH) && (\
    find $(DISTRIB) \( -name .e.pck -o -name '~*' -o -name '*~' -o -name core \) \
    -exec rm {} \; \
- ; cd $(OBJDIR))
+  cd $(OBJDIR)
 
 n_clean:
 	rm -f *.o
-	cd $(VPATH) && (rm -f $(PROGS) $(PROG7) \
-; rm -rf $(OBJDIR)/$(VPATH)/$(DISTRIB) \
-; cd $(OBJDIR) )
+	cd $(VPATH) && (rm -f $(PROGS) $(PROG7); rm -rf $(OBJDIR)/$(VPATH)/$(DISTRIB); cd $(OBJDIR) )
 
 n_distclean: n_clean clean_d
 	cd $(VPATH) && (rm -f *.dir *.pag; cd $(OBJDIR))
 
 make_dir:
-	cd $(VPATH) && (rm -rf $(OBJDIR)/$(VPATH)/$(DISTRIB) \
-; mkdir $(DISTRIB) \
-; mkdir $(DISTRIB)/examples \
-; mkdir $(DISTRIB)/doc \
-; ln -f  \
- $(STERN).[ch] \
- makefile.unx \
- tools.sh \
- Makefile \
- COPYING  \
- README  \
- $(DISTRIB)/. \
-; rm -f  $(DISTRIB)/config.h \
-; ln -f  \
- examples/$(STERN) \
- $(DISTRIB)/examples/. \
-; ln -f  \
- doc/$(STERN) \
- $(DISTRIB)/doc/. \
-; cd $(OBJDIR) )
+	cd $(VPATH) && (rm -rf $(OBJDIR)/$(VPATH)/$(DISTRIB); mkdir $(DISTRIB); mkdir $(DISTRIB)/examples; mkdir $(DISTRIB)/doc; ln -f $(STERN).[ch] makefile.unx tools.sh Makefile COPYING README $(DISTRIB)/.; rm -f $(DISTRIB)/config.h ; ln -f  examples/$(STERN) $(DISTRIB)/examples/. ; ln -f  doc/$(STERN) $(DISTRIB)/doc/.; cd $(OBJDIR) )
 
 n_diff: make_dir clean_d
-	cd $(VPATH) && ( \
- $(MAKEPATCH) org/$(DISTRIB) $(DISTRIB) > $(PATCHF) \
-; gzip -9 -f $(PATCHF) \
-; cd $(OBJDIR) )
+	cd $(VPATH) && ( $(MAKEPATCH) org/$(DISTRIB) $(DISTRIB) > $(PATCHF); gzip -9 -f $(PATCHF); cd $(OBJDIR) )
 
 n_distrib: n_diff
 	-mkdir /tmp/x
diff -ur mars_orig/tools.h mars_new/tools.h
--- mars_orig/tools.h	2003-06-13 12:46:21.000000000 -0400
+++ mars_new/tools.h	2021-01-03 00:01:16.273969399 -0500
@@ -27,7 +27,7 @@
 #define NWBIND    5
 #define NWROUTED  6
 
-extern  FILE *logfile;
+//extern  FILE *logfile;
 extern  void x_x_xfree(char **p);
 extern  int x_x_xnewstr(uint8 **p,  uint8 *s);
 
